crates/ruff_linter/src/rules/flake8_simplify/rules/needless_bool.rs

```diff
@@ -90,12 +118,47 @@ pub(crate) fn needless_bool(checker: &mut Checker, stmt_if: &ast::StmtIf) {
         }] => (
             elif_test,
             elif_body,
-            else_body,
+            else_body.as_slice(),
             TextRange::new(elif_range.start(), else_range.end()),
         ),
+        // if-implicit-else case:
+        // ```python
+        // if x > 0:
+        //     return True
+        // return False
+        // ```
+        [] if checker.settings.preview.is_enabled() => {
+            // Fetching the next sibling is expensive, so do some validation early.
+            if is_one_line_return_bool(if_body).is_none() {
+                return;
+            }
+
+            // Fetch the next sibling statement.
+            let Some(next_stmt) = checker
+                .semantic()
+                .current_statement_parent()
+                .and_then(|parent| traversal::suite(stmt, parent))
+                .and_then(|suite| traversal::next_sibling(stmt, suite))
+            else {
+                return;
+            };
+
+            // If the next sibling is not a return statement, abort.
+            if !next_stmt.is_return_stmt() {
+                return;
+            }
+
+            (
+                if_test.as_ref(),
+                if_body,
+                std::slice::from_ref(next_stmt),
```

Comment:
@ottaviohartman -- I was able to remove the vector allocation here by making every branch return `&[Stmt]` instead of `&Vec<Stmt>`. Then, you can do a nice trick whereby if you have an `&T`, you can create `&[T]` via `std::slice::from_ref`.

Reply:
Awesome - I'll use that in the future.